serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc

```diff
@@ -0,0 +1,212 @@
+= Kubernetes Service Discovery for Serverless Workflow Application Configuration
+:compat-mode!:
+// Metadata:
+:description: Explain what is and how the service discovery works
+:keywords: kogito, workflow, quarkus, serverless, service-discovery, enhanced-service-discovery
+// links
+:quarkus_issue_url: https://github.com/quarkusio/quarkus/issues/27457
+
+
+The Kubernetes Service Discovery allows you to have a static URI that defines a Kubernetes resource that will be used
+to perform HTTP requests. The resource defined in the URI will then be queried in the current Kubernetes cluster and
+translated to a valid URL. The service discovery feature works during the Application startup, it scans all Quarkus
+configuration in search of the following URI pattern. Thus, you should keep in mind that, if the startup time matters to you, then you should consider to use a static URL instead.
+The following sketch demonstrates how this URI is composed:
+
+[source,shell]
+----
+kubernetes:<group>/<version>/<kind>/<namespace>/<name>?<attributeName>=<attributeValue>
+\________/ \_____/ \_______/ \____/ \_________/ \____/ \______________________________/
+  scheme    group   version   kind   namespace   name   additional resource attributes
+            \____________________/                       \__________________________/
+                     GVK                                   Supported values:
+                                                            - port-name={PORT_NUMBER}
+                                                            - labels=label-name=label-value;other-label=other-value
+----
+
+The following scheme values are supported:
+
+* **kubernetes**
+* **openshift**
+* **knative**
+
+
+For the Kubernetes GVK, the following resources are currently supported:
+
+* **v1/service**
+* **serving.knative.dev/v1/service**
+* **v1/pod**
+* **apps/v1/deployment**
+* **apps.openshift.io/v1/deploymentconfig**
+* **apps/v1/statefulset**
+* **route.openshift.io/v1/route**
+* **networking.k8s.io/v1/ingress**
+
+[IMPORTANT]
+====
+The **Kubernetes GVK** (Group, Version and Kind) and the **Name** values are **mandatory** and can't be empty, otherwise, the engine will throw a validation issue and the discovery will not be performed.  +
+The **Namespace** is **optional**. In case of an empty value, the current namespace or context will be used.
+
+For **OpenShift** specific resources, **Deployment Config** or **Routes**, you will be required to add the following dependency in your Application's classpath:
+
+[source,xml,subs="attributes+"]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-openshift-client</artifactId>
+</dependency>
+----
+
+It can be added with link:{quarkus_cli_url}[Quarkus CLI]:
+[source,shell]
+----
+quarkus extension add quarkus-openshift-client
+----
+====
```

Comment:
```suggestion
====

[NOTE]
====
For **OpenShift** specific resources, **Deployment Config** or **Routes**, you will be required to add the following dependency in your Application's classpath:

[source,xml,subs="attributes+"]
----
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-openshift-client</artifactId>
</dependency>
----

It can be added with link:{quarkus_cli_url}[Quarkus CLI]:
[source,shell]
----
quarkus extension add quarkus-openshift-client
----
====
```

Reply:
it was removed.